src/filter.ts

```diff
@@ -5,13 +5,14 @@
  * @param {String} selector
  * @returns {Array}
  */
-export default (nodes: NodeList|HTMLCollection|Array, selector: string): Array<Element> => {
+export default (nodes: NodeList|HTMLCollection|Array<HTMLElement>, selector: string): Array<HTMLElement> => {
   if (!(nodes instanceof NodeList || nodes instanceof HTMLCollection || nodes instanceof Array)) {
     throw new Error('You must provide a nodeList/HTMLCollection/Array of elements to be filtered.')
   }
+  let elArray:Element[] = Array.from(nodes)
```

Comment:
Sorry to be picky, but can you rename `elArray` to just `elements` I think it is more consistent and makes it easier to understand.